tasks/main.yml

```diff
@@ -71,11 +71,20 @@
     force_basic_auth: true
   register: grafana_conf_existing_datasources
 
-- name: Find available datasources
+- name: Template enabled datasources
   set_fact:
-    grafana_conf_available_datasources: "{{ grafana_conf_grafana_datasources|map('extract', grafana_conf_grafana_datasource_definitions)|list }}"
+    grafana_conf_tmp:
+      "{{ grafana_conf_grafana_datasource_definitions[item.key] }}"
+  register: grafana_conf_enabled_datasources
+  with_dict:
+    "{{ grafana_conf_grafana_datasources }}"
```

Comment:
How about building the grafana_conf_enabled_datasources list in one set_fact task?

- set_fact:
    my_var: []
- set_fact:
    my_var: "{{ my_var + [item] }}"
  with_dict: my_dict

I lied. it's still two. You could initialise the list in defaults if you wanted.

Reply:
Seems cleaner - thanks. Ready for re-review.